lib/collectors/global_prometheus_collector.rb

```diff
@@ -0,0 +1,28 @@
+require "prometheus_exporter"
+require "prometheus_exporter/server"
+
+module Collectors
+  class GlobalPrometheusCollector < PrometheusExporter::Server::TypeCollector
+    SECONDS_PER_DAY = 86_400
+
+    def type
+      "locations_api_global"
+    end
+
+    def metrics
+      oldest_os_places_postcode = PrometheusExporter::Metric::Gauge.new("locations_api_oldest_os_places_postcode_age_days", "Days since last update of oldest postcode OS Places postcode")
+      oldest_os_places_postcode.observe(get_oldest_postcode / SECONDS_PER_DAY)
+
+      [oldest_os_places_postcode]
+    end
+
+  private
+
+    def get_oldest_postcode
+      # Cache metric to prevent needless expensive calls to the database
+      Rails.cache.fetch("metrics:oldest_postcode", expires_in: 1.hour) do
+        (Time.zone.now.to_i - Postcode.os_places.order(updated_at: :asc).first.updated_at.to_i)
```

Comment:
I think this is pretty safe because the only way these records get updated is via OS Places API calls (with one exception: bulk CSV import, which is also a reasonable consideration of freshness and is unlikely to happen again anyway).

Reply:
For now yes, agreed. However, if the code changes in the future and the records are updated outside of OS places requests then this is metric may no longer be meaningful. I think relying on the standard timestamp values for harvesting data will probably trip someone up in the future. Don't think I'd classify this as a blocker as we're only at risk of breaking a metric but is a risk and one we'd probably want to consider more if we plan to alert on it 👍